var/spack/repos/builtin/packages/chombo/package.py

```diff
@@ -0,0 +1,140 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+from shutil import copyfile
+import glob
+
+
+class Chombo(MakefilePackage):
+    """The Chombo package provides a set of tools for implementing finite
+       difference and finite-volume methods for the solution of partial
+       differential equations on block-structured adaptively refined
+       logically rectangular (i.e. Cartesian) grids."""
+
+    homepage = "https://commons.lbl.gov/display/chombo"
+    url      = "http://bitbucket.org/drhansj/chombo-xsdk.git"
+
+    # Use whatever path Brian V. and Terry L. agreed upon, but preserve version
+    version('3.2', git='http://bitbucket.org/drhansj/chombo-xsdk.git', commit='71d856c')
+    version('develop', git='http://bitbucket.org/drhansj/chombo-xsdk.git', tag='master')
+#    version('3.2', svn='https://anag-repo.lbl.gov/svn/Chombo/release/3.2')
+
+    variant('mpi', default=True, description='Enable MPI parallel support')
+    variant('hdf5', default=True, description='Enable HDF5 support')
+    variant('dims',
+        default='3',
+        values=('1', '2', '3', '4', '5', '6'),
+        multi=False,
+        description='Number of PDE dimensions [1-6]'
+    )
+
+    patch('hdf5-16api.patch', when='@3.2', level=0)
+    patch('Make.defs.local.template.patch', when='@3.2', level=0)
+
+    depends_on('blas')
+    depends_on('lapack')
+    depends_on('gmake', type='build')
```

Comment:
On thing about @adamjstewart comments here..."make is generally assumed to be on the system...". That kinda sorta fits into this _deep dependency chains_ issue I've been muttering about in several posts/issues.

We need a way for each user of Spack to easily characterize what they think are things that are _generally assumed to be on the system_ such that Spack winds up dealing with dependency chains only as deeply as they expect.

I specifically set a gmake dependency in Chombo because I fear a system without gmake will have problems with it. Most makes are really gmake under the covers. But, this makes Chombo's gmake dependency explicit and I think its best